drake/systems/plants/RigidBodySystem.h

```diff
@@ -388,61 +387,118 @@ class DRAKERBSYSTEM_EXPORT RigidBodySpringDamper
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 };
 
-/** RigidBodySensor
- * @brief interface class for elements which define a sensor which reads the
- * state of a rigid body system
+/** NoiseModel
+ * @brief Represents generalized vector-valued noise
  */
-class DRAKERBSYSTEM_EXPORT RigidBodySensor {
- public:
-  RigidBodySensor(RigidBodySystem* sys, const std::string& name)
-      : sys(sys), name(name) {}
-  virtual ~RigidBodySensor() {}
+template <typename ScalarType, int Dimension>
+class DRAKERBSYSTEM_EXPORT NoiseModel {
+public:
+    virtual Eigen::Matrix<ScalarType, Dimension, 1> generateNoiseVector() = 0;
```

Comment:
This interface assumes that the noise will always be additive. I think you want a method that takes a non-noisy vector by reference and makes it noisy somehow, or takes a vector by const reference and returns a noisy version.